paasta_tools/metrics/metastatus_lib.py

```diff
@@ -322,6 +323,30 @@ def key_func(slave):
     return key_func
 
 
+@functools.total_ordering
+class MultiKey():
+    def __init__(self, attributes):
+        self.attributes = attributes
```

Comment:
if you're writing a hashable datastructure it should be immutable and not dependent on order.  As written, `attributes` has no guarantee of immutability (and is a public mutable member), also `__str__` (and subsequently `__hash__`) aren't resilient to reordering (the mapping type passed in could yield different results for `.items()` dependent on payload).  As such, if inserted into a mapping type it may either:

1. not be retrievable
2. be duplicated by two "keys" that contain the same data but map to two things